src/cljam/io/vcf/reader.clj

```diff
@@ -178,3 +180,41 @@
                     :deep (vcf-util/variant-parser (.meta-info rdr) (.header rdr))
                     :vcf identity)]
      (map parse-fn (read-data-lines (.reader rdr) (.header rdr) kws)))))
+
+(defn- make-lazy-variants [f s]
```

Comment:
`make-lazy-variants` looks roughly equivalent to `mapcat`. Could we replace this with it? Am I missing something?

Reply:
I think name binding is necessary for recursion of lazy sequences.
Is there a better expression?

Reply:
Sorry for putting the comment to a confusing place 🙏 

I meant I asked whether we could replace the invocation to `make-lazy-variants` (that appears at L205) with `mapcat` since they apparently behave in the same way:

```clojure
(make-lazy-variants (juxt dec inc) [1 2 3]) ;=> (0 2 1 3 2 4)
(mapcat (juxt dec inc) [1 2 3]) ;=> (0 2 1 3 2 4)
```

Or do they actually have a difference from some kind of perspective?

Reply:
Thank you. I understand.